### DISCUSSION TITLE
How could I stop mkdocs from converting dashes / hyphens in page filenames into spaces in page titles?

### DISCUSSION THREAD

**Discussion by ryran** [mkdocs/mkdocs#3588] [Q&A]
I really don't understand this behavior whatsoever. I know I can override the title using yaml frontmatter or `# h1 title` on first line, but that defeats the elegance of relying on file structure for auto-nav -- now I have to put the same title in two places (filename and first line) and have to keep them in sync, which is really a bummer.

Please halp with any tips.

**Comment by pawamoy** [comment]
I personally recommend always setting H1 titles, for good hygiene (useful for some plugins such as mkdocs-manpage which generates manual pages). If your file names make good titles, that's pure luck, and soon enough you will need to customize the title for one page. My opinion on this: you write file names (and the whole file tree actually) for the URLs, and you write H1 titles for... titles. Better be explicit about it! My 2 cents :smile: 

Other than that, I don't know if it's possible to prevent MkDocs from converting dashes.

**Comment by kamilkrzyskow** [comment]
Hello @ryran,
if there is no title set manually this is the default behaviour and there is no configuration settings to change it:
https://github.com/mkdocs/mkdocs/blob/e755aaed7ea47348a60495ab364d5483ab90a4a6/mkdocs/structure/pages.py#L256-L259

You could either monkey patch the `Page.title` property with another custom function, but I'm not sure if this would work, as there is the `weak_property` decorator 🤔, but for now I say it's possible. To do that you could use the `on_startup` event in a hook.

Another option would be to restore the `-` yourself, I'm not sure what would be the best event to hook here, but perhaps let's go with `on_page_markdown`, because the file is loaded and processed already, so the default title logic has been executed. Then you set 
`page.title = page.file.name` without the part that replaces the dashes.

https://www.mkdocs.org/user-guide/configuration/#hooks
https://www.mkdocs.org/dev-guide/plugins/#events

Also you mentioned auto navigation, so I expect you mean you don't have a `nav` in the `mkdocs.yml` file and let MkDocs auto generate the navigation tree. If you mean instead you're using the `awesome-pages` plugin then it has it's own logic to get the titles, as MkDocs didn't properly expose one before, perhaps it's time to sync it 🤔 https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/blob/2ff8819072a8fe17b9e80c415e0630f311ca7163/mkdocs_awesome_pages_plugin/navigation.py#L198

**Reply by ryran** [reply]
Thanks Kamil for providing context and for brainstorming some options. I really appreciate it. Maybe someday I'll come back to this and try to figure it out.

I really find it so hard to believe my team are the only folks ever bothered by this. For a team wiki where users set all kinds of titles and titles regularly need to have hyphens & underscores, this seems like such a stupid 1990s anti-pattern / handicap.
